packages/js/product-editor/src/blocks/product-fields/images/edit.tsx

```diff
@@ -92,99 +214,55 @@ export function Edit( {
 					</div>
 				) : (
 					<MediaUploader
-						multipleSelect={ 'add' }
+						value={
+							Array.isArray( propertyValue )
+								? propertyValue.map( ( { id } ) => id )
+								: propertyValue?.id ?? undefined
+						}
+						multipleSelect={ multiple ? 'add' : false }
```

Comment:
This does not seem to be respected for the FileUpload input file where I can still select multiple images: https://github.com/woocommerce/woocommerce/blob/6163cae61befbaf3535a5226d4c1aba2cb7a67ea/packages/js/components/src/media-uploader/media-uploader.tsx#L66 ( it defaults to `multiple={true}`